### ISSUE TITLE
AbstractAssert method for asserting with expected/actual

### ISSUE THREAD

**Issue by kriegfrj** [joel-costigliola/assertj-core#1826]
#### Summary

I think that #1731 was great for a certain class of custom assertions. However, i think it would also be a useful addition to `AbstractAssert` if it had a variant of the `failWithMessage()` method that you could call directly that would populate expected and actual in the generated `AssertionError` (where available). 

#### Example

```java
failWithExpectedActual(expected,  actual, "My message %d", myArg);
```

**Comment by scordio** [comment]
Would it help to use the following?

```java
throw org.assertj.core.internal.Failures.instance().failure(info, shouldBeEqual(actual, expected, info.representation()));
```
As `Failures` belongs to internal APIs, using it in custom assertions is probably not so nice.

**Comment by joel-costigliola** [comment]
I'm ok with the idea, I would actual first and expected after it. `failWithActualAndExpected`.
I would prefer not to recommend suing directly internal APIs.

**Comment by kriegfrj** [comment]
:+1: the internal API is not only a bit 

> Would it help to use the following?
> 
> ```java
> throw org.assertj.core.internal.Failures.instance().failure(info, shouldBeEqual(actual, expected, info.representation()));
> ```
> 
> As `Failures` belongs to internal APIs, using it in custom assertions is probably not so nice.

Thanks @scordio, I tend to agree about it not being nice - but even more than that, if you're using an OSGi environment as I am,  you don't account actually have access to the internal packages.

**Comment by kriegfrj** [comment]
This was implemented by #1833.
